### DISCUSSION TITLE
Ability to proxy Azure OpenAI service

### DISCUSSION THREAD

**Discussion by aavetis** [Helicone/helicone#139] [Roadmap]
Azure has the ability for users to [deploy their own OpenAI service](https://azure.microsoft.com/en-us/products/cognitive-services/openai-service#faq) as part of their cognitive services offerings. By the looks of it, Microsoft will quickly mirror functionality and models as they're made available by OpenAI (e.g. they added chat-turbo the day after it was available). 

See below example from the OpenAI cookbook on how the Azure OpenAI endpoints look. Users give their deployment a `name` and end up with `https://my-service-name.openai.azure.com`. No indication that this url pattern will change, but not sure.

https://github.com/openai/openai-python/blob/main/README.md#microsoft-azure-endpoints

**Potential options:**
- Pass in the Azure endpoint along with the rest of the Helicone params
- Or setup something in the UI for users to specify which API key should go where (either create an Azure service record or send to OpenAI)

**Comment by aavetis** [comment]
updated documentation link, need to provide `api_type`, `api_base` and `api_version`

**Comment by chitalian** [comment]
https://github.com/Helicone/helicone/issues/213
